rules/DeadCode/Rector/ConstFetch/RemovePhpVersionIdCheckRector.php

```diff
@@ -82,195 +91,212 @@ public function run()
-        if (! $node instanceof BinaryOp) {
-            return true;
+        if (! $this->isName($binaryOp->right->name, 'PHP_VERSION_ID')) {
+            return null;
         }
 
-        return $if->cond !== $node;
+        return $this->refactorConstFetch($binaryOp->right, $node, $binaryOp);
     }
 
-    private function processSmaller(ConstFetch $constFetch, Smaller $smaller, If_ $if): ?ConstFetch
+    /**
+     * @return Stmt[]|null
+     */
+    private function processSmaller(ConstFetch $constFetch, Smaller $smaller, If_ $if): ?array
     {
         if ($smaller->left === $constFetch) {
-            return $this->processSmallerLeft($constFetch, $smaller, $if);
+            return $this->processSmallerLeft($smaller, $if);
         }
 
         if ($smaller->right === $constFetch) {
-            return $this->processSmallerRight($constFetch, $smaller, $if);
+            return $this->processSmallerRight($smaller, $if);
         }
 
         return null;
     }
 
-    private function processGreaterOrEqual(ConstFetch $constFetch, GreaterOrEqual $greaterOrEqual, If_ $if): ?ConstFetch
-    {
+    /**
+     * @return Stmt[]|null
+     */
+    private function processGreaterOrEqual(
+        ConstFetch $constFetch,
+        GreaterOrEqual $greaterOrEqual,
+        If_ $if,
+    ): ?array {
         if ($greaterOrEqual->left === $constFetch) {
-            return $this->processGreaterOrEqualLeft($constFetch, $greaterOrEqual, $if);
+            return $this->processGreaterOrEqualLeft($greaterOrEqual, $if);
         }
 
         if ($greaterOrEqual->right === $constFetch) {
-            return $this->processGreaterOrEqualRight($constFetch, $greaterOrEqual, $if);
+            return $this->processGreaterOrEqualRight($greaterOrEqual, $if);
         }
 
         return null;
     }
 
-    private function processSmallerLeft(ConstFetch $constFetch, Smaller $smaller, If_ $if): ?ConstFetch
+    /**
+     * @return null
+     */
+    private function processSmallerLeft(Smaller $smaller, If_ $if)
     {
         $value = $smaller->right;
         if (! $value instanceof LNumber) {
             return null;
         }
 
-        if ($this->phpVersionConstraint >= $value->value) {
+        if ($this->phpVersion >= $value->value) {
             $this->removeNode($if);
         }
 
-        return $constFetch;
+        return null;
     }
 
-    private function processSmallerRight(ConstFetch $constFetch, Smaller $smaller, If_ $if): ?ConstFetch
+    /**
+     * @return Stmt[]|null
+     */
+    private function processSmallerRight(Smaller $smaller, If_ $if): ?array
     {
         $value = $smaller->left;
         if (! $value instanceof LNumber) {
             return null;
         }
 
-        if ($this->phpVersionConstraint >= $value->value) {
-            $this->nodesToAddCollector->addNodesBeforeNode($if->stmts, $if);
-            $this->removeNode($if);
+        if ($this->phpVersion >= $value->value) {
+            return $if->stmts;
```

Comment:
@TomasVotruba it needs ensure no stmts check or it can cause error:

```
There was 1 error:

1) Rector\Tests\DeadCode\Rector\ConstFetch\RemovePhpVersionIdCheckRector\RemovePhpVersionIdCheckRectorTest::test with data set #7 (Symplify\SmartFileSystem\SmartFileInfo Object (...))
Rector\Core\Exception\ShouldNotHappenException: Array of nodes cannot be empty. Ensure "Rector\DeadCode\Rector\ConstFetch\RemovePhpVersionIdCheckRector->refactor()" returns non-empty array for Nodes.

A) Return null for no change:

    return null;

B) Remove the Node:

    $this->removeNode($node);
    return $node;
```

I created PR https://github.com/rectorphp/rector-src/pull/2260 for it.